### PR TITLE
[Keyboard] Add Kaly42

### PR DESCRIPTION
**Pull request by Dwctor**
## Description

I just finished cleaning up the [Kaly42's](https://github.com/Dwctor/Kaly) firmware and would like to get it pushed! To the best of my ability I read all the docs and tried to make this as close as possible to a perfect commit. However, I might have made mistakes as it's my first contribution, so please feel free to tell me what is wrong. I'll work on it ASAP.

ps: everything seemed to work fine without the <keyboard>.c and <keyboard.h> files. Is it fine to leave the code without them?

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kaly/kaly42/info.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
+{
+    "manufacturer": "Dwctor",
+    "keyboard_name": "kaly/kaly42",
+    "maintainer": "Dwctor",
+    "bootloader": "stm32-dfu",
```
Comment (dunk2k) [inline keyboards/kaly/kaly42/info.json:5]:
```suggestion
    "development_board": "blackpill_f401",
```

File: keyboards/kaly/kaly42/info.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                "direct": [
+                    ["B13", "B14", "B15", "A8", "B10", "B1"],
+                    ["A15", "B3", "B4", "B5", "B8", "B9"],
+                    ["B0", "A7", "A6", "A5", "A3", "A4"],
+                    ["A0", "A1", "A2", "NO_PIN", "NO_PIN", "NO_PIN"]
+                ]
+            }
+        }
+    },
+    "processor": "STM32F401",
```
Comment (dunk2k) [inline keyboards/kaly/kaly42/info.json:37]: `processor` defined by `development_board`
```suggestion
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/kaly/kaly42/config.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
```
Comment (dunk2k) [inline keyboards/kaly/kaly42/config.h:33]: non-functional code removed
```suggestion
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by Dwctor** [comment]
@dunk2k I just reviewed and accepted all of your changes! Is anything else amiss?

ps: when I applied your suggestions, did you get notified or is it best practice if I call the reviewer like this when I have done it?

**Comment by dunk2k** [comment]
> ps: when I applied your suggestions, did you get notified or is it best practice if I call the reviewer like this when I have done it?

all those subscribed are notified 😉.

**Comment by dunk2k** [comment]
> Is anything else amiss?

not that I can see 🙂

**Review by dunk2k — APPROVED** [review]

File: keyboards/kaly/kaly42/info.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                ]
+            }
+        }
+    },
+    "url": "https://github.com/Dwctor/Kaly",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0000",
+        "vid": "0xFEED"
+    },
```
Comment (dunk2k) [inline keyboards/kaly/kaly42/info.json:43]:
```suggestion
    },
    "community_layouts": ["split_3x6_3"],
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/kaly/kaly42/info.json (hunk 4 above)
Comment (Dwctor) [reply keyboards/kaly/kaly42/info.json:43]: Sorry about this one! I had added this in another version and forgot to re-add it to this cleaner one, thanks for catching it!

**Review by dunk2k — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**